src/main/scala/snabbdom/VDomProxy.scala

```diff
@@ -14,15 +14,18 @@ object VDomProxy {
 
   import js.JSConverters._
 
-  def attrsToSnabbDom(attributes: Seq[Attribute]): (js.Dictionary[String], js.Dictionary[String]) = {
-    val (attrs, props) = attributes.partition {
-      case (a: Attr) => true
-      case (p: Prop) => false
+  def attrsToSnabbDom(attributes: Seq[Attribute]): (js.Dictionary[String], js.Dictionary[String], js.Dictionary[String]) = {
+    val attrsDict = js.Dictionary[String]()
+    val propsDict = js.Dictionary[String]()
+    val styleDict = js.Dictionary[String]()
+
+    attributes.foreach {
```

Comment:
Can you change this to a fold? :)

Reply:
I really thought about doing this with a fold. But I didn't come up with a way that has the same performance characteristics as my current solution. With building up a triple of lists inside a fold and constructing dictionaries afterwards, we have to iterate twice over every element. We cannot build an immutable triple of dictionaries. And in the end we have to write it into a mutable dictionary anyways: https://github.com/scala-js/scala-js/blob/master/library/src/main/scala/scala/scalajs/js/Dictionary.scala#L59

So I think this is a trade-off between being pseudo-pure and performance. I chose performance. Do you still prefer a fold? ;)